services/web/server/src/simcore_service_webserver/folders/_folders_handlers.py

```diff
@@ -0,0 +1,195 @@
+import functools
+import logging
+
+from aiohttp import web
+from models_library.api_schemas_webserver.folders import (
+    CreateFolderBodyParams,
+    FolderGet,
+    PutFolderBodyParams,
+)
+from models_library.folders import FolderID
+from models_library.rest_ordering import OrderBy, OrderDirection
+from models_library.rest_pagination import PageQueryParameters
+from models_library.users import UserID
+from pydantic import Extra, Field, Json, parse_obj_as, validator
+from servicelib.aiohttp.requests_validation import (
+    RequestParams,
+    StrictRequestParams,
+    parse_request_body_as,
+    parse_request_path_parameters_as,
+    parse_request_query_parameters_as,
+)
+from servicelib.aiohttp.typing_extension import Handler
+from servicelib.mimetype_constants import MIMETYPE_APPLICATION_JSON
+from servicelib.request_keys import RQT_USERID_KEY
+
+from .._constants import RQ_PRODUCT_KEY
+from .._meta import API_VTAG as VTAG
+from ..application_settings_utils import requires_dev_feature_enabled
+from ..login.decorators import login_required
+from ..security.decorators import permission_required
+from ..utils_aiohttp import envelope_json_response
+from . import _folders_api
+from .errors import FolderAccessForbiddenError, FolderNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+def handle_folders_exceptions(handler: Handler):
+    @functools.wraps(handler)
+    async def wrapper(request: web.Request) -> web.StreamResponse:
+        try:
+            return await handler(request)
+
+        except FolderNotFoundError as exc:
+            raise web.HTTPNotFound(reason=f"{exc}") from exc
+
+        except FolderAccessForbiddenError as exc:
+            raise web.HTTPForbidden(reason=f"{exc}") from exc
+
+    return wrapper
+
+
+#
+# folders COLLECTION -------------------------
+#
+
+routes = web.RouteTableDef()
+
+
+class FoldersRequestContext(RequestParams):
+    user_id: UserID = Field(..., alias=RQT_USERID_KEY)
+    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)
+
+
+class FoldersPathParams(StrictRequestParams):
+    folder_id: FolderID
+
+
+class FolderListParams(PageQueryParameters):
+    ...
+
+
+class FolderListWithJsonStrParams(FolderListParams):
+    order_by: Json[OrderBy] = Field(  # pylint: disable=unsubscriptable-object
+        default=OrderBy(field="name", direction=OrderDirection.DESC),
+        description="Order by field (name|description) and direction (asc|desc). The default sorting order is ascending.",
+        example='{"field": "name", "direction": "desc"}',
+        alias="order_by",
+    )
+
+    @validator("order_by", check_fields=False)
+    @classmethod
+    def validate_order_by_field(cls, v):
+        if v.field not in {
+            "name",
+            "description",
+        }:
+            raise ValueError(f"We do not support ordering by provided field {v.field}")
+        return v
+
+    class Config:
+        extra = Extra.forbid
```

Comment:
since a discussion with @pcrespov I am now always wondering if it makes sense to set this Extra.forbid configuration.
If a client passes additional arguments is it really a bad thing? We could just ignore them

Reply:
It's not a problem. This ensures that when a user makes a typo, they fix it right away and do not assume, for example, that something is being filtered when in reality it is not.